pandas/core/frame.py

```diff
@@ -8387,6 +8387,48 @@ def isin(self, values):
                 self.columns,
             )
 
+    def value_counts(self):
+        """
+        The number of times each unique row appears in the DataFrame.
+
+        Returns
+        -------
+        counts : Series
+
+        See Also
+        --------
+        Series.value_counts: Equivalent method on Series.
+
+        Examples
+        --------
+
+        >>> df = pd.DataFrame({'num_legs': [2, 4, 4], 'num_wings': [2, 0, 0]},
+        ...                   index=['falcon', 'dog', 'cat'])
+        >>> df
+                num_legs  num_wings
+        falcon         2          2
+        dog            4          0
+        cat            4          0
+
+        >>> df.value_counts()
+        (4, 0)    2
```

Comment:
Can we return a multi index instead of a tuple index?

Reply:
Done.